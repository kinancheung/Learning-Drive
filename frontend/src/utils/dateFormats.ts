// Date format constant
export const dateFormats = {
  post: 'h:mm a d/MM/yyyy',
};
