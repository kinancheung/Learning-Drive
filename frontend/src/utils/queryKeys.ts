import { postsSearchFilter } from './schema/searchPostsSchema';

export const queryKey = {
  CATEGORIES: 'categories',
  POSTS: (page: number, filter: postsSearchFilter): [string, number, postsSearchFilter] => [
    'posts',
    page,
    filter,
  ],
  PROFILE: (userId?: string): [string, string] => {
    if (!userId) throw new Error();
    return ['profile', userId];
  },
  POST: (postId: string): [string, string] => ['post', postId],
};
