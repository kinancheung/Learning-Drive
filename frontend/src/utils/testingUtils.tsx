import { SnackbarProvider } from 'notistack';
import { PropsWithChildren } from 'react';
import { HelmetProvider } from 'react-helmet-async';
import { MemoryRouter, Location } from 'react-router-dom';

import { AuthContext, AuthContextState } from '../context/AuthContext';

interface composeTestingWrapperOpts {
  initialEntries?: (string | Partial<Location>)[];
  initialIndex?: number;
  authState?: Partial<AuthContextState>;
}

export const composeTestingWrapper = ({
  initialEntries,
  initialIndex,
  authState,
}: composeTestingWrapperOpts = {}) => {
  const Wrapper = ({ children }: PropsWithChildren<{}>) => {
    const defaultState: AuthContextState = {
      signInWithGithub: jest.fn(),
      signInWithGoogle: jest.fn(),
      signInWithFacebook: jest.fn(),
      signOut: jest.fn(),
      ...authState,
    };

    return (
      <HelmetProvider>
        <MemoryRouter initialEntries={initialEntries} initialIndex={initialIndex}>
          <AuthContext.Provider value={defaultState}>
            <SnackbarProvider>{children}</SnackbarProvider>
          </AuthContext.Provider>
        </MemoryRouter>
      </HelmetProvider>
    );
  };
  return Wrapper;
};
