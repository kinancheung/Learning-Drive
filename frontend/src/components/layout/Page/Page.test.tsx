import { render } from '@testing-library/react';

import { Page } from '.';

import { composeTestingWrapper } from '../../../utils/testingUtils';

describe('Page', () => {
  it('renders without error', () => {
    const view = render(<Page title='Global Feed' />, { wrapper: composeTestingWrapper() });

    expect(view.asFragment()).toMatchSnapshot();
  });
});
