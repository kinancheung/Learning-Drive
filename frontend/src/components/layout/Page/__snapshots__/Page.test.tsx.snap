// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders without error 1`] = `
<DocumentFragment>
  <div
    class="css-94a9mv-MuiStack-root"
  >
    <div
      class="MuiBox-root css-j2nflj"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1uiyf5x-MuiToolbar-root-MuiStack-root"
      >
        <h1
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Global Feed
        </h1>
        <div
          class="MuiBox-root css-i9gxme"
        />
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
    </div>
  </div>
</DocumentFragment>
`;
